code/parse/sexp.cpp

```diff
@@ -2702,9 +2712,12 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, i
 
 			case OPF_GOAL_NAME:
 			case OPF_EVENT_NAME:
+			{
 				if (type2 != SEXP_ATOM_STRING)
 					return SEXP_CHECK_TYPE_MISMATCH;
 
+				count = 0;
```

Comment:
How about initializing `count` when declaring it instead?

Reply:
Because `count` is declared at the beginning of `check_sexp_syntax` and could be used in more places than this.